src/Renderer/Camera.js

```diff
@@ -65,60 +80,119 @@ Camera.prototype.setRotation = function setRotation(rotation) {
-        temp.copy(volumes3D.center);
+        tmp.v.copy(volumes3D.center);
     } else {
         throw new Error(`Unsupported volume object ${volumes3D}`);
     }
-    // temp is -center
-    temp.negate();
+    // tmp.v is -center
+    tmp.v.negate();
     // shift the volumes3D toward origin
     tempContainer.copy(volumes3D);
-    tempContainer.translate(temp);
+    tempContainer.translate(tmp.v);
 
     // modify position: substract camera.position and add box3d.min
-    tempMatrix.elements[12] -= visibilityTestingOffset.x + temp.x;
-    tempMatrix.elements[13] -= visibilityTestingOffset.y + temp.y;
-    tempMatrix.elements[14] -= visibilityTestingOffset.z + temp.z;
+    tmp.matrix.elements[12] -= visibilityTestingOffset.x + tmp.v.x * tmp.scale.x;
+    tmp.matrix.elements[13] -= visibilityTestingOffset.y + tmp.v.y * tmp.scale.y;
+    tmp.matrix.elements[14] -= visibilityTestingOffset.z + tmp.v.z * tmp.scale.z;
 }
 
 Camera.prototype.isBox3DVisible = function isBox3DVisible(box3d, matrixWorld) {
-    _prepareMatrix(matrixWorld, this._visibilityTestingOffset, box3d, tempBox3d);
+    _prepareMatrix(matrixWorld, this._visibilityTestingOffset, box3d, tmp.box3d);
 
-    localViewMatrix.multiplyMatrices(this._viewMatrix, tempMatrix);
-    frustum.setFromMatrix(localViewMatrix);
-    return frustum.intersectsBox(tempBox3d);
+    tmp.viewMatrix.multiplyMatrices(this._viewMatrix, tmp.matrix);
+    tmp.frustum.setFromMatrix(tmp.viewMatrix);
+    return tmp.frustum.intersectsBox(tmp.box3d);
 };
 
 Camera.prototype.isSphereVisible = function isSphereVisible(sphere, matrixWorld) {
-    _prepareMatrix(matrixWorld, this._visibilityTestingOffset, sphere, tempSphere3d);
+    _prepareMatrix(matrixWorld, this._visibilityTestingOffset, sphere, tmp.sphere3d);
 
-    localViewMatrix.multiplyMatrices(this._viewMatrix, tempMatrix);
-    frustum.setFromMatrix(localViewMatrix);
-    return frustum.intersectsSphere(tempSphere3d);
+    tmp.viewMatrix.multiplyMatrices(this._viewMatrix, tmp.matrix);
+    tmp.frustum.setFromMatrix(tmp.viewMatrix);
+    return tmp.frustum.intersectsSphere(tmp.sphere3d);
 };
 
 Camera.prototype.box3DSizeOnScreen = function box3DSizeOnScreen(box3d, matrixWorld) {
-    _prepareMatrix(matrixWorld, this._visibilityTestingOffset, box3d, tempBox3d);
-    tempMatrix.premultiply(this._viewMatrix);
+    _prepareMatrix(matrixWorld, this._visibilityTestingOffset, box3d, tmp.box3d);
+    tmp.matrix.premultiply(this._viewMatrix);
+
+    return tmp.box3d.applyMatrix4(tmp.matrix);
+};
+
+Camera.prototype.box3DContourOnScreen = (function _box3DContourOnScreen() {
+    const points = [
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+    ];
+    return function box3DContourOnScreen(box3d, matrixWorld, clampToScreen = true) {
+        if (box3d.isEmpty()) {
```

Comment:
The jsfiddle method is useful when you only need to render the contour of a solid object.

The `box3dContourOnScreen` returns the geometry of the contour - and when extended to handle any kind of geometry it will work with things like pointcloud for instance.